src/main/java/org/folio/fqm/repository/ResultSetRepository.java

```diff
@@ -104,11 +109,28 @@ public List<Map<String, Object>> getResultSet(UUID entityTypeId, Fql fql, List<S
     }
 
     Condition condition = FqlToSqlConverterService.getSqlCondition(fql.fqlCondition(), entityType);
+    // TODO: might want to put next 5 lines in its own method in EntityTypeFlatteningService
+    //   also is it even necessary?
+    String finalWhereClause = condition.toString();
+    for (EntityTypeSource source : entityType.getSources()) {
+      String toReplace = ":" + source.getAlias();
+      String alias = "\"" + source.getAlias() + "\"";
+      finalWhereClause = finalWhereClause.replace(toReplace, alias);
+    }
     var fieldsToSelect = getSqlFields(entityType, fields);
-    var sortCriteria = hasIdColumn(entityType) ? field(ID_FIELD_NAME) : DSL.noField();
+    String idFieldWithAlias = "";
+    String idColumnName = entityType
+      .getColumns()
+      .stream()
+      .filter(EntityTypeColumn::getIsIdColumn)
+      .findFirst()
+      .orElseThrow()
```

Comment:
This should throw a specific exception, or at least a generic one with a clear error message